test/test-migrate-export.sh

```diff
@@ -0,0 +1,432 @@
+  assert_pointer "refs/heads/my-feature" "a.md" "$md_feature_oid" "30"
+
+  git lfs migrate export --include="*.md, *.txt"
+
+  [ ! $(assert_pointer "refs/heads/master" "a.md" "$md_oid" "140") ]
+  [ ! $(assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "120") ]
+  [ ! $(assert_pointer "refs/heads/master" "b.md" "$b_md_oid" "160")]
+  assert_pointer "refs/heads/my-feature" "a.md" "$md_feature_oid" "30"
+
+  # b.md should be pruned as no pointer exists to reference it
+  refute_local_object "$b_md_oid" "160"
+
+  # Other objects should not be pruned as they're still referenced in `feature`
+  # by pointers
+  assert_local_object "$md_oid" "140"
+  assert_local_object "$txt_oid" "120"
+  assert_local_object "$md_feature_oid" "30"
+
+  master="$(git rev-parse refs/heads/master)"
+  feature="$(git rev-parse refs/heads/my-feature)"
+
+  master_attrs="$(git cat-file -p "$master:.gitattributes")"
+  feature_attrs="$(git cat-file -p "$feature:.gitattributes")"
+
+  echo "$master_attrs" | grep -q "*.md text !filter !merge !diff"
+  echo "$master_attrs" | grep -q "*.txt text !filter !merge !diff"
+
+  [ ! $(echo "$feature_attrs" | grep -q "*.md text !filter !merge !diff") ]
+  [ ! $(echo "$feature_attrs" | grep -q "*.txt text !filter !merge !diff") ]
+)
+end_test
+
+begin_test "migrate export (with remote)"
+(
+  set -e
+
+  setup_single_remote_branch_tracked
+
+  git push origin master
+
+  md_oid="$(calc_oid "$(cat a.md)")"
+  txt_oid="$(calc_oid "$(cat a.txt)")"
+
+  assert_pointer "refs/heads/master" "a.md" "$md_oid" "50"
+  assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "30"
+
+  assert_pointer "refs/remotes/origin/master" "a.md" "$md_oid" "50"
+  assert_pointer "refs/remotes/origin/master" "a.txt" "$txt_oid" "30"
+
+  # Flush the cache to ensure all objects have to be downloaded
+  rm -rf .git/lfs/objects
+
+  git lfs migrate export --everything --include="*.md, *.txt"
+
+  [ ! $(assert_pointer "refs/heads/master" "a.md" "$md_oid" "50") ]
+  [ ! $(assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "30") ]
+
+  [ ! $(assert_pointer "refs/remotes/origin/master" "a.md" "$md_oid" "50") ]
+  [ ! $(assert_pointer "refs/remotes/origin/master" "a.txt" "$txt_oid" "30") ]
+
+  # All pointers have been exported, so all objects should be pruned
+  refute_local_object "$md_oid" "50"
+  refute_local_object "$txt_oid" "30"
+
+  master="$(git rev-parse refs/heads/master)"
+  master_attrs="$(git cat-file -p "$master:.gitattributes")"
+
+  echo "$master_attrs" | grep -q "*.md text !filter !merge !diff"
+  echo "$master_attrs" | grep -q "*.txt text !filter !merge !diff"
+)
+end_test
+
+begin_test "migrate export (include/exclude args)"
+(
+  set -e
+
+  setup_single_local_branch_tracked
+
+  md_oid="$(calc_oid "$(cat a.md)")"
+  txt_oid="$(calc_oid "$(cat a.txt)")"
+
+  assert_pointer "refs/heads/master" "a.txt" "$txt_oid" "120"
+  assert_pointer "refs/heads/master" "a.md" "$md_oid" "140"
+
+  git lfs migrate export --include="*" --exclude="a.md"
```

Comment:
I'm not sure if this would be the place to put it, but I think that the `--include="*"` flag should _only_ be the default if no `--include` or `--exclude` argument is given, since I think writing patterns against an implicit `--include="*"` would be confusing for users.

Reply:
Regarding my comment above, I was suggesting that if we decide on a default value of `"*"` for `--include`, it should _only_ be the default when neither `--include=` nor `--exclude=` is given.